### ISSUE TITLE
model.eval() seems not work well

### ISSUE THREAD

**Issue by mapleneverfade** [zhanghang1989/PyTorch-Encoding#35]
It's very kind of your work ! There's still something i want for your help!
My code for model and criterion parallel like this:
        model = encoding.parallel.ModelDataParallel(model,device_ids=[0,1,2])
        criterion = encoding.parallel.CriterionDataParallel(criterion,device_ids=[0,1,2])
Training process is going well, but when i turn to model.eval(), i got explosion of loss.
![image](https://user-images.githubusercontent.com/19401116/38480319-977bc56e-3bf7-11e8-917c-2ef3875dd91f.png)
Is there something wrong with model.eval()?

**Comment by zhanghang1989** [comment]
Are you training with evaluation mode?

**Comment by mapleneverfade** [comment]
I set model.train() before every epoch of  loader.
![image](https://user-images.githubusercontent.com/19401116/38591848-706cdd62-3d6b-11e8-8290-b74a3eb89204.png)

**Comment by zhanghang1989** [comment]
Thx for clarifying! What is the mIoU you were expecting? For example, what is the mIoU when using standard BatchNorm at 1st epoch?

**Comment by mapleneverfade** [comment]
![image](https://user-images.githubusercontent.com/19401116/38593463-0b8f383c-3d74-11e8-866f-365fba2ad444.png)

**Comment by zhanghang1989** [comment]
Am I missing something. You said, "i turn to model.eval(), i got explosion of loss."
What does it mean? We typically do not calculate loss in evaluation mode.

**Comment by mapleneverfade** [comment]
Maybe I didn't clear it out. When I use standard BatchNorm I get normal loss both in train mode and eval mode.
While using syn-bn I get norm loss during training but explode in eval mode, it  means  there maybe 
something wrong with syn-bn when turned it to eval mode.
I test the syn-bn model in model.eval(), got this
![image](https://user-images.githubusercontent.com/19401116/38653682-760acb84-3e3e-11e8-9acb-13bb3eff2248.png)
Calculating eval loss is not what i expect to, I just want to figure out what causes the test IoU  to be zero.
Or is  there something I misused？

**Comment by zhanghang1989** [comment]
Please checkout the PyTorch compatible Synchronized Cross-GPU [encoding.nn.BatchNorm2d](http://hangzh.com/PyTorch-Encoding/syncbn.html) and the [example](https://github.com/zhanghang1989/PyTorch-SyncBatchNorm).

**Comment by zhanghang1989** [comment]
@mapleneverfade The sycBN works the same as standard BN in eval mode. Please try the new PyTorch DataParallel compatible version.

**Comment by zhanghang1989** [comment]
Hi, have you checked out the example https://github.com/zhanghang1989/PyTorch-SyncBatchNorm

**Comment by zhanghang1989** [comment]
Hi @mapleneverfade , do you still have the problem? I still couldn't get it why do you calculate loss in eval mode
